ray/operator/base/kustomization.yaml

```diff
@@ -1,9 +1,28 @@
----
-kind: Kustomization
-apiVersion: kustomize.config.k8s.io/v1beta1
-resources:
-  - ray-operator-serviceaccount.yaml
-  - ray-operator-role.yaml
-  - ray-operator-rolebinding.yaml
-  - ray-operator-deployment.yaml
-  - ray-custom-resources.yaml
+# Adds namespace to all resources.
+namespace: ray-system
```

Comment:
I don't think we should block merging the PR on this, but I think we should think through a way to un-hardcode the namespace for this.

Reply:
@accorvin :+1:

The operator lives in its own namespace, so I don't think it should be installed inside of say, the ODH namespace. `ray-system` is the default name used by the KubeRay Project.  Is there an issue with it spawning another namespace?

Reply:
Using the `namespace:` in `kustomization.yaml` is the preferred method since it propagates to every namespace reference in the manifest and there are plugins we could use to add hooks that modify that functionality

Reply:
Yeah so I think the only issue here is that we also create the namespace and the name is hardcoded there, so even if someone changed the namespace passed to kustomize it would still create the `ray-system` namespace.

Reply:
So that is an issue with the use [kustomize v3.2.0](https://github.com/opendatahub-io/opendatahub-operator/blob/master/go.mod#L61) in the current operator code where it does not run the `namespace:` transformer on `kind: Namespace` objects.

That issue will be resolved when we update the kustomize version in the operator to kustomize >= [v3.8.3](https://github.com/kubernetes-sigs/kustomize/pull/2953) this month.

@accorvin I'll defer to you if you want remove the Namespace creation.

Reply:
I think if we keep the namespace creation here then we should include basic instructions for how to change/override the namespace. IMO it's better to make namespace creation a prerequisite step that the user must do.